src/main/java/uk/gov/companieshouse/api/accounts/controller/CompanyAccountController.java

```diff
@@ -50,10 +53,26 @@ public ResponseEntity createCompanyAccount(@Valid @RequestBody CompanyAccount co
                 .getAttribute(AttributeName.TRANSACTION.getValue());
 
         String requestId = request.getHeader("X-Request-Id");
-        ResponseObject result = companyAccountService
-                .createCompanyAccount(companyAccount, transaction, requestId);
-        return apiResponseMapper
-                .map(result.getStatus(), result.getData(), result.getErrorData());
+
+        ResponseEntity responseEntity;
+        try {
+            CompanyAccount companyAccountInserted = companyAccountService
+                    .createCompanyAccount(companyAccount, transaction, requestId);
+            responseEntity = apiResponseMapper
+                    .map(ResponseStatus.SUCCESS_CREATED, companyAccountInserted);
```

Comment:
we should be using response.getStatus() here rather than having controllers deciding which ResponseStatus to use. 

setting the ResponseStatus is the responsibility of the service, the controller uses this to map to a HTTP response code (via the apiResponseMapper utility)